### PR TITLE
Update sha2, sha3 and curve25519-dalek dependencies

### PR DESCRIPTION
**Pull request by remoun**
Also merge in some updates from upstream `w3f/master`.

References https://github.com/mobilecoinfoundation/curve25519-dalek/pull/2

Supports https://github.com/mobilecoinfoundation/mobilecoin/issues/1326

### CODE REVIEW

**Review by cbeck88 — DISMISSED** [review]

**Review by jcape — DISMISSED** [review]
LGTM pending curve patch location update.

**Comment by remoun** [comment]
PTAL, I've updated the patched curve dep to use the MCF fork.

File: Cargo.toml [hunk 1]
```diff
@@ -127,3 +122,5 @@ getrandom = ["rand_core/getrandom"]
 # See https://github.com/rust-lang/cargo/issues/9210
 # and https://github.com/w3f/schnorrkel/issues/65#issuecomment-786923588
 
+[patch.crates-io]
+curve25519-dalek = { git = "https://github.com/mobilecoinfoundation/curve25519-dalek.git", rev = "8791722e0273762552c9a056eaccb7df6baf44d7" }
```
Comment (remoun) [inline Cargo.toml:126]: This can be removed when https://github.com/dalek-cryptography/curve25519-dalek/pull/383 is published

**Review by jcape — DISMISSED** [review]

**Comment by remoun** [comment]
PTAL, I had to force-push after signing the commits.

**Review by jcape — APPROVED** [review]

**Pull request merged**